### PR TITLE
[bazel] Run //quality:buildifier_fix

### PR DESCRIPTION
**Pull request by alphan**
Signed-off-by: Alphan Ulusoy <alphan@google.com>

### CODE REVIEW

**Comment by alphan** [comment]
@milesdai, I'm not sure why I see this in my local clone, maybe these slipped when the buildifier check was disabled for some reason?

**Review by drewmacrae — APPROVED** [review]
This seems good, all these edits fix things.

I thought we ignore some of these files, but I'm not sure of it. I'm trying to work out why CI isn't telling us about these issues.

**Comment by alphan** [comment]
> 

Thanks!

**Comment by drewmacrae** [comment]
I think CI has exceptions that effect only //quality:buildifier_check and not //quality:buildifier_fix, but I'm not sure.

**Comment by drewmacrae** [comment]
buildifier_check just runs on the dif, while buildifier_fix runs on everything (or the arguments youu pass)

**Comment by milesdai** [comment]
> buildifier_check just runs on the dif, while buildifier_fix runs on everything (or the arguments youu pass)

Is this still true anymore? We did a refactoring of the quality rules recently, and I think the check runs on the whole repo now: https://cs.opensource.google/opentitan/opentitan/+/master:quality/BUILD.bazel;l=111

There may be an issue with the globbing though. I think Bazel's glob function doesn't find subpackages, so maybe this is only checking top-level directories. Let me look into this a bit more.

File: third_party/rust/crates/crates.bzl [hunk 1]
```diff
@@ -213,347 +213,176 @@ def raze_fetch_remote_crates(
...
-        zerocopy__0_5_0=None,
-
-        zerocopy_derive__0_3_1=None,
-
-        zeroize__1_4_3=None,
-
-        zeroize_derive__1_3_2=None,
-
-    ):
+        CoreFoundation_sys__0_1_4 = None,
```
Comment (cfrantz) [inline third_party/rust/crates/crates.bzl:216]: This file is autogenerated and should be added to the list of excludes.

**Comment by cfrantz** [comment]
The current buildifier_check rule has an error.  Unfortunately, the `glob()` is not crossing package boundaries.  We're probably going to have to rewrite the implementation of the rule to work more like `clang_format_check`.

**Comment by milesdai** [comment]
> The current buildifier_check rule has an error. Unfortunately, the `glob()` is not crossing package boundaries. We're probably going to have to rewrite the implementation of the rule to work more like `clang_format_check`.

Got it - I can take a look at redoing this rule

**Comment by alphan** [comment]
@cfrantz @milesdai would you like me to close this and create an issue instead? I think this PR would have served its purpose either way.

**Comment by milesdai** [comment]
Yeah it looks like there are some complications with this rule. I'm filing an issue to discuss some more, but go ahead and merge this PR.

**Comment by alphan** [comment]
> Yeah it looks like there are some complications with this rule. I'm filing an issue to discuss some more, but go ahead and ~merge~ this PR. Edit: I meant _close_ this PR.

Thanks @milesdai!

**Pull request closed without merging**